rcl/test/rcl/test_security_directory.cpp

```diff
@@ -49,33 +48,40 @@ static int unsetenv_wrapper(const char * var_name)
 class TestGetSecureRoot : public ::testing::Test
 {
 protected:
+  rcl_allocator_t allocator;
+  char * root_path;
+
   void SetUp() override
   {
+    // Reset rcutil error global state in case a previously
+    // running test has failed.
+    rcl_reset_error();
+
     // Always make sure the variable we set is unset at the beginning of a test
     unsetenv_wrapper(ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME);
     unsetenv_wrapper(ROS_SECURITY_NODE_DIRECTORY_VAR_NAME);
     unsetenv_wrapper(ROS_SECURITY_LOOKUP_TYPE_VAR_NAME);
+    allocator = rcl_get_default_allocator();
+    root_path = "";
   }
 };
 
 TEST_F(TestGetSecureRoot, failureScenarios) {
-  rcl_allocator_t allocator = rcl_get_default_allocator();
-  ASSERT_EQ(rcl_get_secure_root(TEST_NODE_NAME, TEST_NODE_NAMESPACE, &allocator),
+  ASSERT_EQ(rcl_get_secure_root(TEST_NODE_NAME, TEST_NODE_NAMESPACE, allocator),
```

Comment:
Isn't it this leaking?

Reply:
all tests in failureScenarios return NULL. All memory allocated by calling `rcl_get_secure_root` are deallocated by the allocator before `rcl_get_secure_root` return NULL.

Reply:
Ok so it's very minor but we'll be leaking if the test fails.
This is also using ASSERT which is probably not warranted here, EXPECT should probably be used instead.
=> https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html

If you want, you could solve this problem through a small utility function but again, not super important.